src/fipp/engine.cljc

```diff
@@ -229,10 +235,9 @@
 (defn pprint-document [document options]
   (let [options (merge {:width 70} options)]
     (->> (serialize document)
-         (eduction
-           annotate-rights
-           (annotate-begins options)
-           (format-nodes options))
+         (sequence (comp annotate-rights
```

Comment:
Does this significantly hurt performance by forcing a lazy sequence to be realized with all of the intermediate data structures that entails?